pkgs/os-specific/bsd/freebsd/by-name/libc.nix

```diff
@@ -0,0 +1,263 @@
+  ];
+
+  postPatch = ''
+    substituteInPlace $COMPONENT_PATH/Makefile --replace '.include <src.opts.mk>' ""
+
+    substituteInPlace $BSDSRCDIR/include/paths.h \
+        --replace '/usr/lib/i18n' '${builtins.placeholder "out"}/lib/i18n' \
+        --replace '/usr/share/i18n' '${builtins.placeholder "out"}/share/i18n'
+  '';
+
+  nativeBuildInputs = [
+    buildPackages.bsdSetupHook buildFreebsd.freebsdSetupHook
+    buildFreebsd.bmakeMinimal
+    buildFreebsd.install
+
+    buildPackages.flex buildPackages.byacc buildFreebsd.gencat buildFreebsd.rpcgen
+    buildFreebsd.mkcsmapper buildFreebsd.mkesdb
+  ];
+  buildInputs = [ include csu ];
+  env.NIX_CFLAGS_COMPILE = toString [
+    "-B${csu}/lib"
+    # These are supposed to have _RTLD_COMPAT_LIB_SUFFIX so we can get things like "lib32"
+    # but that's unnecessary
+    "-DSTANDARD_LIBRARY_PATH=\"${builtins.placeholder "out"}/lib\""
+    "-D_PATH_RTLD=\"${builtins.placeholder "out"}/libexec/ld-elf.so.1\""
+  ];
+
+  makeFlags = [
+    "STRIP=-s" # flag to install, not command
+    # lib/libc/gen/getgrent.c has sketchy cast from `void *` to enum
+    "MK_WERROR=no"
+  ];
+
+  MK_SYMVER = "yes";
+  MK_SSP = "yes";
+  MK_NLS = "yes";
+  MK_ICONV = "yes";
+  MK_NS_CACHING = "yes";
+  MK_INET6_SUPPORT = "yes";
+  MK_HESIOD = "yes";
+  MK_NIS = "yes";
+  MK_HYPERV = "yes";
+  MK_FP_LIBC = "yes";
+
+  MK_TCSH = "no";
+  MK_MALLOC_PRODUCTION = "yes";
+
+  MK_TESTS = "no";
+  MACHINE_ABI = "";
+  MK_DETECT_TZ_CHANGES = "no";
+  MK_MACHDEP_OPTIMIZATIONS = "yes";
+  MK_ASAN = "no";
+  MK_UBSAN = "no";
+
+  NO_FSCHG = "yes";
+
+  preBuild = lib.optionalString (stdenv.hostPlatform.isx86_32) ''
+    make -C $BSDSRCDIR/lib/libssp_nonshared $makeFlags
+    make -C $BSDSRCDIR/lib/libssp_nonshared $makeFlags install
+  '';
+
+  postInstall = ''
+    pushd ${include}
+    find . -type d -exec mkdir -p $out/\{} \;
+    find . \( -type f -o -type l \) -exec cp -pr \{} $out/\{} \;
+    popd
+
+    pushd ${csu}
+    find . -type d -exec mkdir -p $out/\{} \;
+    find . \( -type f -o -type l \) -exec cp -pr \{} $out/\{} \;
+    popd
+
+    mkdir $BSDSRCDIR/lib/libcompiler_rt/i386
+    make -C $BSDSRCDIR/lib/libcompiler_rt $makeFlags
+    make -C $BSDSRCDIR/lib/libcompiler_rt $makeFlags install
+
+    make -C $BSDSRCDIR/lib/libgcc_eh $makeFlags
+    make -C $BSDSRCDIR/lib/libgcc_eh $makeFlags install
+
+    ln -s $BSDSRCDIR/lib/libc/libc.so.7 $BSDSRCDIR/lib/libc/libc.so  # not sure
+    mkdir $BSDSRCDIR/lib/libgcc_s/i386
+    make -C $BSDSRCDIR/lib/libgcc_s $makeFlags
+    make -C $BSDSRCDIR/lib/libgcc_s $makeFlags install
+
+    NIX_CFLAGS_COMPILE+=" -B$out/lib"
+    NIX_CFLAGS_COMPILE+=" -I$out/include"
+    NIX_LDFLAGS+=" -L$out/lib"
+
+    make -C $BSDSRCDIR/lib/libc_nonshared $makeFlags
+    make -C $BSDSRCDIR/lib/libc_nonshared $makeFlags install
+
+    mkdir $BSDSRCDIR/lib/libmd/sys
+    make -C $BSDSRCDIR/lib/libmd $makeFlags
+    make -C $BSDSRCDIR/lib/libmd $makeFlags install
```

Comment:
Can't this be deduplicated a lot by using ``for a in b c``?

Reply:
It can be, but I find it a lot easier to enumerate the options fully, especially when there are neighboring lines which ought to fit into the loop but have overrides applied.